### DISCUSSION TITLE
"No text opener found"

### DISCUSSION THREAD

**Discussion by vbd** [sxyazi/yazi#1607] [1. Q&A]
I selected some files and tried to bulk rename them. After pressing `r` I get "No text opener found".
I'm on Windows with yazi 0.3. Any idea what todo?

**Comment by sxyazi** [comment]
What does `yazi --debug` say?

**Comment by vbd** [comment]
I think this part is relevant:

    ...
    Variables
        SHELL              : None
        EDITOR             : None
    ...

Hm... should I try to set env variable EDITOR to vim.exe?

**Reply by AnirudhG07** [reply]
yes try that.

**Reply by vbd** [reply]
Just done. No "No text opener found" message. :)
But...
Just created a test directory containg `file1` and `file2`.
In yazi I selected both files with `space` and hit `r`.
Vim starts, but the buffer is empty.

**Reply by sxyazi** [reply]
Paste full `yazi --debug`

**Reply by sxyazi** [reply]
> Vim starts, but the buffer is empty

What's the filename shown at the bottom of Vim? If you open that file in another editor, does it have any content?

**Comment by vbd** [comment]
Yazi
    Version: 0.3.0 (7a380c2 2024-08-01)
    Debug  : false
    OS     : windows-x86_64 (windows)

Ya
    Version: 0.3.0

Emulator
    Emulator.via_env: ("", "")
    Emulator.via_csi: Ok(Unknown([]))
    Emulator.detect : Unknown([])

Adapter
    Adapter.matches: Chafa

Desktop
    XDG_SESSION_TYPE: None
    WAYLAND_DISPLAY : None
    DISPLAY         : None

SSH
    shared.in_ssh_connection: false

WSL
    /proc/sys/fs/binfmt_misc/WSLInterop: false

Variables
    SHELL              : None
    EDITOR             : Some("vim.exe")
    YAZI_FILE_ONE      : Some("c:\\Program Files\\Git\\usr\\bin\\file.exe")
    YAZI_CONFIG_HOME   : None
    ZELLIJ_SESSION_NAME: None

Text Opener
    default: Some(Opener { run: "vim.exe %*", block: true, orphan: false, desc: "vim", for_: None, spread: true })
    block  : Some(Opener { run: "vim.exe %*", block: true, orphan: false, desc: "vim", for_: None, spread: true })

tmux
    TMUX   : false
    Version: program not found

Dependencies
    file             : 5.45
    ueberzugpp       : program not found
    ffmpegthumbnailer: program not found
    magick           : program not found
    fzf              : 0.54.3
    fd               : 10.1.0
    rg               : 13.0.0
    chafa            : program not found
    zoxide           : 0.9.4
    7z               : 24.06
    7zz              : program not found
    jq               : 1.7

**Reply by AnirudhG07** [reply]
Your version is outdated, download the latest 0.3.2 as well

**Reply by sxyazi** [reply]
> block : Some(Opener { run: "vim.exe %", block: true, orphan: false, desc: "vim", for_: None, spread: true })

Did you reopen Yazi after setting `block=true`?

**Reply by vbd** [reply]
Yes I restarted it. I switched just to 0.3.2.

**Reply by vbd** [reply]
EDIT: After switching to 0.3.2 the vim buffer is still empty. New debug and log file see below.

**Comment by vbd** [comment]
Yazi
    Version: 0.3.2 (ddb8ce5 2024-08-28)
    Debug  : false
    OS     : windows-x86_64 (windows)

Ya
    Version: 0.3.2 (ddb8ce5 2024-08-28)

Emulator
    Emulator.via_env: ("", "")
    Emulator.via_csi: Ok(Unknown([]))
    Emulator.detect : Unknown([])

Adapter
    Adapter.matches: Chafa

Desktop
    XDG_SESSION_TYPE: None
    WAYLAND_DISPLAY : None
    DISPLAY         : None

SSH
    shared.in_ssh_connection: false

WSL
    /proc/sys/fs/binfmt_misc/WSLInterop: false

Variables
    SHELL              : None
    EDITOR             : Some("vim.exe")
    VISUAL             : Some("vim.exe")
    YAZI_FILE_ONE      : Some("c:\\Program Files\\Git\\usr\\bin\\file.exe")
    YAZI_CONFIG_HOME   : None

Text Opener
    default: Some(Opener { run: "vim.exe %*", block: true, orphan: false, desc: "vim", for_: None, spread: true })
    block  : Some(Opener { run: "vim.exe %*", block: true, orphan: false, desc: "vim", for_: None, spread: true })

Multiplexers
    TMUX               : false
    tmux version       : program not found
    ZELLIJ_SESSION_NAME: None
    Zellij version     : program not found

Dependencies
    file             : 5.45
    ueberzugpp       : program not found
    ffmpegthumbnailer: program not found
    magick           : program not found
    fzf              : 0.54.3
    fd               : 10.1.0
    rg               : 13.0.0
    chafa            : program not found
    zoxide           : 0.9.4
    7z               : 24.06
    7zz              : program not found
    jq               : 1.7

**Reply by vbd** [reply]
2024-09-04T12:13:33.208722Z  WARN yazi_adapter::emulator: [Adapter] No special environment variables detected
    at yazi-adapter\src\emulator.rs:70

  2024-09-04T12:13:33.208753Z  WARN yazi_adapter::emulator: [Adapter] Unknown TERM_PROGRAM: 
    at yazi-adapter\src\emulator.rs:84

  2024-09-04T12:13:33.208755Z  WARN yazi_adapter::emulator: [Adapter] Unknown TERM: 
    at yazi-adapter\src\emulator.rs:92

  2024-09-04T12:13:33.209094Z  WARN yazi_adapter::adapter: [Adapter] Could not identify XDG_SESSION_TYPE
    at yazi-adapter\src\adapter.rs:101

  2024-09-04T12:13:33.209130Z  WARN yazi_adapter::adapter: [Adapter] Falling back to chafa
    at yazi-adapter\src\adapter.rs:113

  2024-09-04T12:13:33.210490Z  WARN yazi_adapter::ueberzug: ueberzugpp spawning failed: program not found
    at yazi-adapter\src\ueberzug.rs:87

**Reply by sxyazi** [reply]
That's weird, I didn't notice any obvious issues. Does editing files directly in Yazi with Vim work?

**Reply by vbd** [reply]
What do you mean with directly? I can open and edit files in vim fromy within yazi and this just works fine.

**Reply by vbd** [reply]
Ok. Found it. I use a kind of "hack" to start gvim with vim.exe. Changing this hack to console vim.exe it works. The trouble is my "hack". So sorry for the time and work I have caused you.

**Reply by AnirudhG07** [reply]
Great to know it's solved! 
Cause technically if you are able to run vim through yazi, then for renaming also it should work just fine

**Reply by vbd** [reply]
Thanks for all your help. Everything works fine. 
Just removed my "hack" to avoid future problems.

**Reply by AnirudhG07** [reply]
We should add in the issues another field, are you using an alias for some command or not, LOL XD

**Reply by vbd** [reply]
At least that might have made me think before I started the discussion. Sorry again for that.
